libs/sso-client/src/lib/sso-client.ts

```diff
@@ -266,3 +288,84 @@ function splitPath(path: string): string[] {
 
   return split;
 }
+
+async function initializeFakeUser(request: NextRequest) {
+  const access_token = request.cookies.get(
+    getEnv().Hibiscus.Cookies.accessTokenName
+  );
+  const refresh_token = request.cookies.get(
+    getEnv().Hibiscus.Cookies.refreshTokenName
+  );
+  const supabase = createSupabaseServiceClient();
+
+  if (
+    access_token == null ||
+    refresh_token == null ||
+    (await supabase.auth.getUser(access_token)).data.user == null
+  ) {
+    let {
+      data: { user, session },
+    } = await supabase.auth.signInWithPassword({
+      email: FAKE_USER_EMAIL,
+      password: FAKE_USER_PASSWORD,
+    });
+
+    if (user == null || session == null) {
+      ({
+        data: { user },
+      } = await supabase.auth.admin.createUser({
+        email: FAKE_USER_EMAIL,
+        password: FAKE_USER_PASSWORD,
+        email_confirm: true,
+      }));
+
+      if (user != null) {
+        ({
+          data: { user, session },
+        } = await supabase.auth.signInWithPassword({
+          email: FAKE_USER_EMAIL,
+          password: FAKE_USER_PASSWORD,
+        }));
+      }
+    }
+
+    if (user != null && session != null) {
+      await supabase.from('user_profiles').insert({
+        user_id: user.id,
+        email: user.email,
+        first_name: 'Hack',
+        last_name: 'SC',
+        // Default role = HACKER
+        role: Object.keys(HibiscusRole).indexOf(HibiscusRole.HACKER) + 1,
+      });
+
+      request.cookies.set(
+        getEnv().Hibiscus.Cookies.accessTokenName,
+        session.access_token,
+        {
+          path: '/',
+          domain: getEnv().Hibiscus.AppURL.baseDomain,
+          maxAge: Number.parseInt(getEnv().Hibiscus.Cookies.maxAge),
+          sameSite: 'lax',
+        }
+      );
+      request.cookies.set(
+        getEnv().Hibiscus.Cookies.refreshTokenName,
+        session.refresh_token,
+        {
+          path: '/',
+          domain: getEnv().Hibiscus.AppURL.baseDomain,
+          maxAge: Number.parseInt(getEnv().Hibiscus.Cookies.maxAge),
+          sameSite: 'lax',
+        }
+      );
+    }
+  }
+}
+
+function createSupabaseServiceClient(): SupabaseClient {
+  return createClient(
+    getEnv().Hibiscus.Supabase.apiUrl,
+    getEnv().Hibiscus.Supabase.serviceKey
+  );
+}
```

Comment:
you can do this with the `HibiscusSupabaseClient`'s `setOptions({useServiceKey: true})` so you shouldn't need this